integration_test/ops_agent_test.go

```diff
@@ -465,74 +465,95 @@ func TestHTTPRequestLog(t *testing.T) {
 			"status":        200,
 		}
 
-		// Write a log with an http request in the message to cloud logging and immediately query it.
-		writeAndQueryLog := func(httpRequestKey string, logId string) *cloudlogging.Entry {
-			logBody := map[string]interface{}{
-				"logId":        logId,
-				httpRequestKey: httpRequestBody,
-			}
-			logBytes, err := json.Marshal(logBody)
-			if err != nil {
-				t.Fatalf("could not marshal test log: %v", err)
-			}
-			err = gce.UploadContent(
-				ctx,
-				logger,
-				vm,
-				strings.NewReader(string(logBytes)+"\n"),
-				logPath)
-			if err != nil {
-				t.Fatalf("error writing log line: %v", err)
-			}
+		// Log with HTTP request data nested under "logging.googleapis.com/httpRequest".
+		// const newHTTPRequestKey = confgenerator.HttpRequestKey
+		const newHTTPRequestKey = "logging.googleapis.com/http_request"
+		const newHTTPRequestLogId = "new_request_log"
+		newLogBody := map[string]interface{}{
+			"logId":           newHTTPRequestLogId,
+			newHTTPRequestKey: httpRequestBody,
+		}
+		newLogBytes, err := json.Marshal(newLogBody)
+		if err != nil {
+			t.Fatalf("could not marshal new test log: %v", err)
+		}
+
+		// Log with HTTP request data nested under "logging.googleapis.com/http_request".
+		// const oldHTTPRequestKey = "logging.googleapis.com/http_request"
+		const oldHTTPRequestKey = confgenerator.HttpRequestKey
+		const oldHTTPRequestLogId = "old_request_log"
+		oldLogBody := map[string]interface{}{
+			"logId":           oldHTTPRequestLogId,
+			oldHTTPRequestKey: httpRequestBody,
+		}
+		oldLogBytes, err := json.Marshal(oldLogBody)
+		if err != nil {
+			t.Fatalf("could not marshal old test log: %v", err)
+		}
+
+		// Write both logs to log source file at the same time.
+		err = gce.UploadContent(
+			ctx,
+			logger,
+			vm,
+			strings.NewReader(fmt.Sprintf("%s\n%s\n", string(newLogBytes), string(oldLogBytes))),
+			logPath)
+		if err != nil {
+			t.Fatalf("error writing log line: %v", err)
+		}
 
-			entry, err := gce.QueryLog(
+		queryLogById := func(logId string) (*cloudlogging.Entry, error) {
+			return gce.QueryLog(
 				ctx,
 				logger.ToMainLog(),
 				vm,
 				"mylog_source",
 				time.Hour,
 				fmt.Sprintf("jsonPayload.logId=%q", logId),
 				gce.QueryMaxAttempts)
-			if err != nil {
-				t.Fatalf("could not find written log with id %s: %v", logId, err)
```

Comment:
You could keep this message in the common function via `fmt.Errorf`, and use it as `t.Fatal(err.Error())`…